pyproject.toml

```diff
@@ -0,0 +1,117 @@
+[build-system]
+requires = ["setuptools>=40.8.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "dspy-ai"
+version = "2.0.8"
+description = "DSPy"
+readme = "README.md"
+authors = [{name = "Omar Khattab", email = "okhattab@stanford.edu"}]
+license = {text = "MIT License"}
+requires-python = ">=3.9, <3.12"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3", # removed 3.8
+    "Programming Language :: Python :: 3.9",
+]
+dependencies = [
+    "backoff~=2.2.1",
+    "joblib~=1.3.2",
+    "openai~=0.28.1",
+    "pandas~=2.1.1",
+    "regex~=2023.10.3",
+    "ujson~=5.8.0",
+    "tqdm~=4.66.1",
+    "datasets~=2.14.6",
+    "requests~=2.31.0",
+    "optuna~=3.4.0",
+]
+
+[project.optional-dependencies]
+pinecone = ["pinecone-client~=2.2.4"]
+qdrant = ["qdrant-client~=1.6.2", "fastembed~=0.1.0"]
+chromadb = ["chromadb~=0.4.14"]
+marqo = ["marqo"]
+docs = [
+    "sphinx>=4.3.0",
+    "furo>=2023.3.27",
+    "docutils<0.17",
+    "m2r2",
+    "myst-parser",
+    "myst-nb",
+    "sphinx-autobuild",
+    "sphinx_rtd_theme",
+    "pydantic<2.0.0",
+    "autodoc_pydantic",
+    "sphinx-reredirects>=0.1.2",
+    "sphinx-automodapi==0.16.0"
+]
+
+[project.urls]
+homepage = "https://github.com/stanfordnlp/dsp"
+
+[tool.poetry]
+name = "dspy"
+version = "2.0.8"
+description = "DSPy"
+authors = ["Omar Khattab <okhattab@stanford.edu>"]
+license = "MIT"
+readme = "README.md"
+homepage = "https://github.com/stanfordnlp/dsp"
+documentation = "https://dspy-ai.readthedocs.io"
+repository = "https://github.com/stanfordnlp/dsp"
+keywords = ["dsp", "ai", "signal processing"]
```

Comment:
this has a few hallucinated pieces (signal processing and dspy-ai.readthedocs.io don't exist here haha).